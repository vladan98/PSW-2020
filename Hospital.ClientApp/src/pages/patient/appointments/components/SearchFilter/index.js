import moment from "moment"
import React from 'react';
import { Grid, FormControl, Typography, InputLabel, MenuItem, Select, Button, TextField } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
    paper: {
        marginTop: theme.spacing(8),
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
    },
    form: {
        width: '100%', // Fix IE 11 issue.
        marginTop: theme.spacing(1),
    },
    submit: {
        margin: theme.spacing(3, 0, 2),
    }
}));


const SearchFilter = ({ register }) => {
    const classes = useStyles();

    return (
        <>
            <Typography component="h2" variant="h6">
                Search Appointments
            </Typography>
            <Grid container justify="center" spacing={2}>
                <Grid item>
                    <TextField
                        label="From"
                        type="date"
                        data-cy="schedule-startDate-input"
                        {...register("startDate")}
                        defaultValue={moment().format("YYYY-MM-DD")}
                        variant="outlined"
                        margin="normal"
                        fullWidth
                        autoFocus
                    />
                </Grid>
                <Grid item>
                    <TextField
                        label="To"
                        data-cy="schedule-endDate-input"
                        type="date"
                        {...register("endDate")}
                        defaultValue={moment().add(3, "days").format("YYYY-MM-DD")}
                        variant="outlined"
                        margin="normal"
                        fullWidth
                        autoFocus
                    />
                </Grid>
                <Grid item>
                    <FormControl variant="outlined" margin="normal" fullWidth >
                        <InputLabel>Type</InputLabel>
                        <Select
                            defaultValue={"0"}
                            data-cy="schedule-typeOfAppointment-input"
                            label="Type"
                            {...register("typeOfAppointment")}
                        >
                            <MenuItem data-cy="schedule-typeOfAppointment-input-0" value={"0"}>Examination</MenuItem>
                            <MenuItem data-cy="schedule-typeOfAppointment-input-1" value={"1"}>Surgery</MenuItem>
                        </Select>
                    </FormControl>
                </Grid>
                <Grid item>
                    <FormControl variant="outlined" margin="normal" fullWidth >
                        <InputLabel >Priority</InputLabel>
                        <Select
                            data-cy="schedule-priority-input"
                            defaultValue={"0"}
                            label="Priority"
                            {...register("priority")}
                        >
                            <MenuItem data-cy="schedule-priority-input-0" value={"0"}>None</MenuItem>
                            <MenuItem data-cy="schedule-priority-input-1" value={"1"}>Doctor</MenuItem>
                            <MenuItem data-cy="schedule-priority-input-2" value={"2"}>Date</MenuItem>
                        </Select>
                    </FormControl>
                </Grid>
                <Grid item>
                    <Button
                        data-cy="schedule-submit"
                        type="submit"
                        fullWidth
                        variant="contained"
                        color="primary"
                        className={classes.submit}
                        margin="normal"
                    >
                        Search
                    </Button>
                </Grid>
            </Grid>
        </>
    );
}

export default SearchFilter