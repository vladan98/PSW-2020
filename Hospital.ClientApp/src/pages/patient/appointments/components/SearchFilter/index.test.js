import SearchFilter from "."
import render from "../../../../../helpers/renderComponent"

describe("SearchFilter", () => {
    it("should render page with search form", () => {
        const { container } = render(<SearchFilter register={() => { }} referrals={[]} />)

        expect(container).toMatchSnapshot()
    })
})