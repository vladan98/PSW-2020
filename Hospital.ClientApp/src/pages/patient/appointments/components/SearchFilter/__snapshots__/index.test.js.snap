// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter should render page with search form 1`] = `
<div>
  <h2
    class="MuiTypography-root MuiTypography-h6"
  >
    Search Appointments
  </h2>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        data-cy="schedule-startDate-input"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink="true"
        >
          From
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            type="date"
            value="2021-07-02"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-6 PrivateNotchedOutline-legendNotched-7"
            >
              <span>
                From
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        data-cy="schedule-endDate-input"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-focused Mui-focused"
          data-shrink="true"
        >
          To
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            type="date"
            value="2021-07-05"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-6 PrivateNotchedOutline-legendNotched-7"
            >
              <span>
                To
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink="true"
        >
          Type
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          data-cy="schedule-typeOfAppointment-input"
        >
          <div
            aria-haspopup="listbox"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
            role="button"
            tabindex="0"
          >
            Examination
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value="0"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-6 PrivateNotchedOutline-legendNotched-7"
            >
              <span>
                Type
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink="true"
        >
          Priority
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          data-cy="schedule-priority-input"
        >
          <div
            aria-haspopup="listbox"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
            role="button"
            tabindex="0"
          >
            None
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value="0"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-6 PrivateNotchedOutline-legendNotched-7"
            >
              <span>
                Priority
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-3 MuiButton-containedPrimary MuiButton-fullWidth"
        data-cy="schedule-submit"
        margin="normal"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Search
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
